pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,313 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			egressIpStr := fmt.Sprintf("%v", egressIp)
+			err = client.Default().Dynamic().Resource(gvrCloudPrivateIPConfig).Delete(ctx, egressIpStr, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+		eipNamespaceStr := fmt.Sprintf("%v", eipNamespace)
+		nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespaceStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		nnsResourceVersion := nns.Object["metadata"].(map[string]interface{})["resourceVersion"]
+		nnsNetId := nns.Object["netid"]
+		netName := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+
+		netnamespace := &uns.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "network.openshift.io/v1",
+				"kind":       "NetNamespace",
+				"metadata": map[string]interface{}{
+					"name":            netName,
+					"resourceVersion": nnsResourceVersion,
+				},
+				"netid":     nnsNetId,
+				"netname":   netName,
+				"egressIPs": egressIps,
+			},
+		}
+		_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, netnamespace, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList.Items {
+		if _, ok := node.Labels["k8s.ovn.org/egress-assignable"]; ok {
+			// parse out node's subnet from egressip annotation
+			egressIpConfigParsed := strings.SplitAfter(node.Annotations["cloud.network.openshift.io/egress-ipconfig"], "\"ipv4\":\"")
+			nodeSubnet := egressIpConfigParsed[1]
+			nodeSubnet = nodeSubnet[:strings.IndexByte(nodeSubnet, '"')]
+			nodeSubnetArr := []string{nodeSubnet}
+
+			hostIpParsed := strings.SplitAfter(node.Annotations["k8s.ovn.org/host-addresses"], "[\"")
+			hostIp := hostIpParsed[1]
+			hostIp = hostIp[:strings.IndexByte(hostIp, '"')]
+
+			hsn, err := client.Default().Dynamic().Resource(gvrHostSubnet).Get(ctx, node.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			hsnResourceVersion := hsn.Object["metadata"].(map[string]interface{})["resourceVersion"]
+			hsnSubnet := hsn.Object["subnet"]
+
+			hostSubnet := &uns.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion":  "network.openshift.io/v1",
+					"kind":        "HostSubnet",
+					"egressCIDRs": nodeSubnetArr,
+					"host":        node.Name,
+					"hostIP":      hostIp,
+					"metadata": map[string]interface{}{
+						"name":            node.Name,
+						"resourceVersion": hsnResourceVersion,
+					},
+					"subnet": hsnSubnet,
+				},
+			}
+			_, err = client.Default().Dynamic().Resource(gvrHostSubnet).Update(ctx, hostSubnet, metav1.UpdateOptions{})
```

Comment:
Ditto. Shall we use Patch instead of Update here?

Reply:
See my [previous reply](https://github.com/openshift/cluster-network-operator/pull/1536/files/8630b5f0d5bdfdfcaa5e4a5af1dfb9490b283554#r959866164).